### DISCUSSION TITLE
Fail to build batch ell structure on gpu

### DISCUSSION THREAD

**Discussion by AbdelhadiKara** [ginkgo-project/ginkgo#1575] [Q&A] [answered]
Hello, i am trying to build batch ell matrix in order to use batched bicgstab. the implementation i have works perfectly on cpu and provide expected results. 
For gpu (nvidia )  kokkos views are used to provide data to gko. I have seen that there is an example [here](https://github.com/ginkgo-project/ginkgo/blob/v1.7.0/examples/kokkos_assembly/kokkos_assembly.cpp)
and uses **gko::device_matrix_data**. Is there a same approach to do with batch ell ?

**Comment by pratikvn** [comment, marked as answer]
If you have the data and that already is in a layout that works for Ginkgo batched matrices and solvers, then you can just use a gpu executor instead of the cpu executor and that should work transparently. Instead of providing the cpu data pointer, you would provide the pointer to the data obtained from the kokkos view.

**Reply by AbdelhadiKara** [reply]
Thank you for the answer. It is actually the case, the matrices structure and and x,b multivector are all built using kokkos views. with particular attention to make the corresponding objects ie ginkgo::array  as views also.

**Reply by pratikvn** [reply]
Yes, then instead of using a reference or the omp executor, you can just use the cuda executor, and that should be sufficient. If you share your code, I can help if you are facing any issues.
